pkg/controllers/canary_controller.go

```diff
@@ -142,13 +142,15 @@ func (r *CanaryReconciler) Reconcile(parentCtx gocontext.Context, req ctrl.Reque
 	if canaryForStatus.Status.Replicas != canary.Spec.Replicas {
 		if canary.Spec.Replicas == 0 {
 			canaryJobs.Unschedule(canary.GetPersistedID())
-			if err := db.SuspendCanary(ctx, canary.GetPersistedID(), true); err != nil {
+
+			if err := r.suspendCanary(ctx, canary, true); err != nil {
```

Comment:
I don't think we should update the suspend annotation based on spec.replicas. If a manual suspend annotation is applied and then the canary is scaled to 0, and then back up again that manually applied annotation will be lost -  spec.replicas=0 should be equivalent to schedule: @never

Reply:
The problem with relying on `spec.Replicas` is that by default it's always `0`.